File: app/src/main/kotlin/com/fishweeb/practical/StateBase.java
package com.fishweeb.practical;

import android.graphics.Canvas;
import android.view.SurfaceView;

public interface StateBase
{
        String GetName();

        void OnEnter(SurfaceView _view);
        void OnExit();
        void Render(Canvas _canvas);
        void Update(float _dt);
}

File: app/src/main/java/edu/androidgroupproject/StateBase.kt
package edu.androidgroupproject

import android.graphics.Canvas
import android.view.SurfaceView

interface StateBase {
    fun GetName(): String

    fun OnEnter(_view: SurfaceView)
    fun OnExit()
    fun Render(_canvas: Canvas)
    fun Update(_dt: Float)
}
